### PR TITLE
Generate bootstrap bindata with streams, templates and quickstarts

### PR DESCRIPTION
**Pull request by csrwng**
To be used by 'oc cluster up' aka 'oc start'

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees @smarterclayton ptal

File: hack/verify-examples.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+if ! output=`OUTPUT_ROOT=${TMP_GENERATED_EXAMPLES_DIR} ${OS_ROOT}/hack/build-examples.sh 2>&1`
+then
+	echo "FAILURE: Generation of fresh bindata failed:"
+	echo "$output"
+  exit 1
+fi
+
+echo "Diffing current examples bindata against freshly generated bindata"
+ret=0
+diff -Nap "${OS_ROOT}/pkg/examples/bindata.go" "${TMP_GENERATED_EXAMPLES_DIR}/pkg/examples/bindata.go" &> /dev/null || ret=$?
```
Comment (smarterclayton) [inline hack/verify-examples.sh]: I don't like "examples" as the word here.  It's data related to bootstrapping a cluster, or initializing it, or represents system defaults.  Examples doesn't really work there.

Comment (csrwng) [inline hack/verify-examples.sh]: bootstrap sound better?

**Comment by csrwng** [comment]
@smarterclayton better?

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
Pullthrough [test]

**Comment by bparees** [comment]
> Pullthrough

@smarterclayton does pullthrough have retry logic for when it has issues reaching dockerhub/etc?

**Comment by bparees** [comment]
lgtm but i'm far from comfortable in this space.

**Comment by smarterclayton** [comment]
There was one spot left, Maceij was hoping that the hub would be fixed
but it hasn't been yet.

**Comment by smarterclayton** [comment]
So why did this pass if verify is working and you didn't check in the files?

**Comment by csrwng** [comment]
I did check in the bindata. Github doesn't expand it because it's too big.

**Comment by smarterclayton** [comment]
So you did, it disappeared.  LGTM [merge]

**Comment by smarterclayton** [comment]
GitHub flaked due to some sort of general outage, try again [merge]

**Comment by smarterclayton** [comment]
[merge] AWS

**Comment by smarterclayton** [comment]
[merge]

**Comment by csrwng** [comment]
[merge]

**Pull request merged**